### PR TITLE
Follow ups for accessibility of announcements slider

### LINKED ISSUE #2494: [Slideshow] Stop auto-rotating slides when user interacts with its controls
## Describe the current behavior
Slideshow section: Auto-rotates even when users interact with the previous/next buttons and the pagination controls.

## Describe the expected behavior
**Better usability/accessibility: Once a user has explicitly activated one of these previous/next buttons, the auto-rotating functionality should stop.** If users are currently reading/interacting with content in the carousel, and the slide that they are currently on is advanced and replaced with new content the experience will be disorienting, as they may not realize the reason why the content they were currently on suddenly disappeared.

## Version information (Dawn, browsers and operating systems)

- Dawn Version: 7.0.1
- Chrome Version 108.0.5359.124
- macOS Version 13.1

### PR DESCRIPTION
**Pull request by eugenekasimov**
### PR Summary: 

This PR:

- changes settings of auto-rotation for announcements. Min speed is 5 sec, max is 10 sec, off for default.
- prevents auto-rotation for mobile.
- stops auto-rotation after first interaction with an arrow button.

### Why are these changes introduced?

Fixes #2494 

### What approach did you take?

Using JS I prevent auto-rotation for mobile and also stop it if a user interacts with an arrow button.

### Testing steps/scenarios

- [ ] Add a few announcements to the announcement bar.
- [ ] In the settings set `Auto-rotate slides`.
- [ ] Change the screen size to mobile and make sure that the slider stops rotation, then resize the screen back and make sure the slider continues rotating. (! while testing this don't interact with arrow buttons since this will stop the slider).
- [ ] Interact with an arrow button and make sure that the slider stops rotating. Resize the screen to mobile and back and make sure it doesn't resume rotating. 

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139851431958/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by stufreen — COMMENTED** [review]
Are we sure we want to apply the changes to every Slideshow, or is it just for Announcement bar?

If we only want the changes on Announcement bar then it may be time to create another custom component. With the animation changes we will probably need a different component anyway.

**Review by melissaperreault — APPROVED** [review]
LGTM! 🎉

File: assets/global.js [hunk 1]
```diff
@@ -701,6 +701,7 @@ class SlideshowComponent extends SliderComponent {
     super();
     this.sliderControlWrapper = this.querySelector('.slider-buttons');
     this.enableSliderLooping = true;
+    this.arrowButtonWasInteracted = false;
```
Comment (ludoboludo) [inline assets/global.js:704]: I'm kind of wondering if we should wrap some of that code in a conditional. 

Something like: 
```javascript
if (this.classList.contains('announcement-bar') {
  ... run the logic you're adding
```
Or create a variable that is `this.announcementBarSlideshow = this.classList.contains('announcement-bar')`

**Review by ludoboludo — COMMENTED** [review]
I agree with Stu that adding a feature specific logic in a general component class seem a bit off. 
But for now we can go with that. We can create an issue and look into a refactor later.

File: assets/global.js [hunk 2]
```diff
@@ -712,11 +713,17 @@ class SlideshowComponent extends SliderComponent {
...
 
+    this.desktopLayout = window.matchMedia('(min-width: 750px)');
     this.reducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)');
+
     this.reducedMotion.addEventListener('change', () => {
       if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
     });
 
+    this.desktopLayout.addEventListener('change', (event) => {
+      if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
```
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/global.js:724]: Hey @ludoboludo , thanks for the feedback. This was coded on a purpose to cover the case when for some reason you resize your browser from desktop to tablet/mobile and back. I guess it's really a rare case though. If we are okay not to cover it, then you are right, we don't need this here.

File: assets/global.js (hunk 1 above)
Comment (eugenekasimov) [reply assets/global.js:704]: I think that kind of condition I already have in line 744 - 751, but ideally we want to separate all additional logic into a new component or extension for the slideshow component. Let's do it as a next step.

File: assets/global.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eugenekasimov) [reply assets/global.js:724]: It seems good to me, I think I could keep all code related to the announcement bar together in constructor () and add this condition you mentioned, this way we will improve it a little bit.

File: assets/global.js (hunk 1 above)
Comment (eugenekasimov) [reply assets/global.js]: [it's done](https://github.com/Shopify/dawn/pull/2580/commits/00d23e0bf98fe7beb6aa47d33c5b1e02699bd14c#diff-47a8b2f1172e03da15783b379f0e007d8e3dfd27cbd9142adf9e32c4b9591079R715-R728).

File: assets/global.js [hunk 3]
```diff
@@ -712,10 +712,27 @@ class SlideshowComponent extends SliderComponent {
...
-    });
+    if (this.querySelector('.announcement-bar-slider')) {
+      this.announcementBarArrowButtonWasClicked = false;
+
+      this.desktopLayout = window.matchMedia('(min-width: 750px)');
+      this.reducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)');
+
+      this.reducedMotion.addEventListener('change', () => {
+        if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
+      });
```
[the remark this answers is not in the document]
Comment (stufreen) [reply assets/global.js:723]: +1 to moving the `reducedMotion` stuff outside of `if (this.querySelector('.announcement-bar-slider'))` like it was before.

**Review by stufreen — APPROVED** [review]
LGTM with a couple minor changes

File: assets/global.js [hunk 4]
```diff
@@ -712,10 +712,26 @@ class SlideshowComponent extends SliderComponent {
...
+
+      this.desktopLayout.addEventListener('change', (event) => {
+        if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
+      });
```
Comment (ludoboludo) [inline assets/global.js:727]: I think you can do the same as we did with the buttons on those `change` events 🤔 

```javascript
[this.desktopLayout, this.reducedMotion].forEach((mediaQuery) => {
  mediaQuery.addEventListener('change', () => {
    if (this.slider.getAttribute('data-autoplay') === 'true') this.setAutoPlay();
  });
});
```

**Review by ludoboludo — APPROVED** [review]
One small suggestion otherwise looks good 👍

File: assets/global.js (hunk 4 above)
Comment (eugenekasimov) [reply assets/global.js:727]: Thanks Ludo.

**Pull request merged**